src/models/FileSystem.js

```diff
@@ -222,6 +222,12 @@ export class FileSystem {
   async lstat(filename) {
     try {
       const stats = await this._lstat(filename)
+      // For non-browser (local) scenarios regular git 'add' command might be used to write the index. Therefore we need to have the exact nanoseconds (see. normalizeStats.js SecondsNanoseconds ).
```

Comment:
> So I think the issue occurs if users use their repository with git and isomorphic-git. If you stage a file with git and then call statusMatrix with isomorphic-git then you compare the git lstat with the isomorphic-git lstat and there can be differences. And if you have differences then you throw away the cache and get a performance issue.

@dead-end thats right, and is mentioned here. if you use native git the timestamp (nanoseconds) is more accurate than the timestamp ('faked' nanoseconds by using ms * 1000000) in isomorphic-git. (this is also the reason why the git index is broken after isomorphic-git tries to rebuild it, becuase it uses the 'faked' nanoseconds) 

Thats why we need to differentiate between browser and local. With `bigint: true` isomorphic-git is able to create the same timestamp (nanoseconds) as native git does.

Reply:
@jayree You have written: 

_(this is also the reason why the git index is broken after isomorphic-git tries to rebuild it, becuase it uses the 'faked' nanoseconds)_

Can you explain how the faked nanos lead to the broken index file? We have a lot of issues with this and I would like to understand it.

Reply:
@dead-end with the old behaviour nanoseconds were calculated based on `ms`. 

From a file with `mtimeMs: 1683216016493.005` 
you'll get `1683216016s` and `493.0048828125ms` --> `493004882.8125ns` 
Due to the numeric data types, floating point values are used, which is not the case in the BigInt data structure.
If you use the `mtimeNs: 1683216016493004997n` (bigint) value
you'll get `1683216016s` and `493004997ns`. These are the vaules git uses in the index. 

The old logic fixing the #1245 compared `493004882.8125ns` with `493004997ns` which caused Isomorphic-git to think the index is broken telling isomorphic-git to rebuild the index, but with `1683216016s` `493004882.8125ns` wich doesn't match the real `ns` timestamp of the file.